packages/jaeger-ui/src/model/link-patterns.tsx

```diff
@@ -139,6 +139,35 @@ function callTemplate(template: ProcessedTemplate, data: any) {
   return template.template(data);
 }
 
+export function computeTraceLinks(linkPatterns: ProcessedLinkPattern[], trace: Trace) {
+  const result: { url: string; text: string }[] = [];
+  const validKeys = Object.keys(trace)
+    // @ts-ignore
```

Comment:
I spent the morning reading why `Object.keys(t: T)` does not return `Array<keyof T>` and still fundamentally disagree that TS won't make a compile time check here due to potential runtime abuse...

Anyway, `@ts-ignore` can be removed by adding a typecast to `Object.keys(trace)` as such:
`(Object.keys(trace) as (keyof Trace)[]).filter(key => typeof trace[key] === 'string' || trace[key] === 'number')`